src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -192,23 +230,80 @@ public void setNumberOfInstances(int numberOfInstances) {
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
-			}
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));				
```

Comment:
Here is another implicit assumption: Let us assume that promregator was deployed as CF app "promregator", then a network policy was required as well:

``` 
cf add-network-policy promregator --destination-app apptoscrape --protocol tcp --port 9090
``` 

==> another thing for documentation

Additionally, the route which has also to be mapped (see other comment below) also needs to be entered in the `preferredRouteRegex` of that application.

Reply:
Yes the network policies are important and need to be created to allow promregator to speak to anything internally. This will have to go into the docs.

I will start an additional file on internal routing.